internal/resource/url.go

```diff
@@ -473,6 +478,29 @@ func (f *Fetcher) fetchFromS3WithCreds(ctx context.Context, dest s3target, input
 	return nil
 }
 
+type S3Retryer struct {
+	client.DefaultRetryer
+}
+
+// MaxRetries returns the configured number of NumMaxRetries, defaults to 3
+func (r S3Retryer) MaxRetries() int {
```

Comment:
I think I tend to prefer the way they're handling `NumMaxRetries` upstream where it's expected that the user will specify the value they want to the object rather than having a default built-in like this. Maybe let's add a package level `DefaultNumMaxRetries = 3` & just use that when we configure the `S3Retryer` object?